openedx/core/djangoapps/user_api/accounts/helpers.py

```diff
@@ -0,0 +1,56 @@
+"""
+Helper functions for the accounts API.
+"""
+import hashlib
+
+from django.conf import settings
+from django.core.files.storage import FileSystemStorage, get_storage_class
+
+PROFILE_IMAGE_SIZES = {
+    'full': 500,
+    'large': 120,
+    'medium': 50,
+    'small': 30
+}
+PROFILE_IMAGE_FORMAT = 'jpg'
+
+
+def get_profile_image_url_for_user(user, size):
+    """Return the URL to a user's profile image for a given size.
+    Note that based on the value of
+    django.conf.settings.PROFILE_IMAGE_DOMAIN, the URL may be relative,
+    and in that case the caller is responsible for constructing the full
+    URL.
+
+    If the user has not yet uploaded a profile image, return the URL to
+    the default edX user profile image.
+
+    Arguments:
+        user (django.auth.User): The user for whom we're generating a
+        profile image URL.
+
+    Returns:
+        string: The URL for the user's profile image.
+
+    Raises:
+        ValueError: The caller asked for an unsupported image size.
+    """
+    if size not in PROFILE_IMAGE_SIZES.values():
```

Comment:
This is a little tricky... I could separate those values out into a list or tuple, but then I'd have something like...

```python
PROFILE_IMAGE_SIZES = (500, 120, 50, 30)
PROFILE_IMAGE_SIZES_MAP = {
    'full': PROFILE_IMAGE_SIZES[0],
    'large': PROFILE_IMAGE_SIZES[1],
    'medium': PROFILE_IMAGE_SIZES[2],
    'small': PROFILE_IMAGE_SIZES[3]
}
```

... since we still need a way of mapping keys for the API response to URLs containing those pixel sizes. A side-effect of that approach is that overriding these settings becomes more cumbersome when testing.

Or were you suggesting something like this, where the list is still computed, but at module load time?:
```python
PROFILE_IMAGE_SIZES_MAP = {
    'full': PROFILE_IMAGE_SIZES[0],
    'large': PROFILE_IMAGE_SIZES[1],
    'medium': PROFILE_IMAGE_SIZES[2],
    'small': PROFILE_IMAGE_SIZES[3]
}
PROFILE_IMAGE_SIZES = PROFILE_IMAGE_SIZES_MAP.values()
```